`r-package (windows-latest, MINGW, R 3.6, cran)` CI job fails

Current `master` branch logs:
```
2022-02-27T14:10:10.2027009Z [command]C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command "$ErrorActionPreference = 'Stop' ; try { Add-Type -AssemblyName System.IO.Compression.FileSystem } catch { } ; [System.IO.Compression.ZipFile]::ExtractToDirectory('D:\a\_temp\ab3810a3-16dc-4fed-a31e-9ae5a0411491', 'D:\a\_temp\2b709202-9ee1-412b-b85f-35881214033e')"
2022-02-27T14:10:15.7306408Z ##[group]Run $env:BUILD_SOURCESDIRECTORY = $env:GITHUB_WORKSPACE
2022-02-27T14:10:15.7307083Z [36;1m$env:BUILD_SOURCESDIRECTORY = $env:GITHUB_WORKSPACE[0m
2022-02-27T14:10:15.7307414Z [36;1m$env:TOOLCHAIN = "MINGW"[0m
2022-02-27T14:10:15.7307668Z [36;1m$env:R_VERSION = "3.6"[0m
2022-02-27T14:10:15.7307934Z [36;1m$env:R_BUILD_TYPE = "cran"[0m
2022-02-27T14:10:15.7308212Z [36;1m$env:COMPILER = "MINGW"[0m
2022-02-27T14:10:15.7308464Z [36;1m$env:GITHUB_ACTIONS = "true"[0m
2022-02-27T14:10:15.7308735Z [36;1m$env:TASK = "r-package"[0m
2022-02-27T14:10:15.7309032Z [36;1m& "$env:GITHUB_WORKSPACE/.ci/test_windows.ps1"[0m
2022-02-27T14:10:15.7357874Z shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". {0}"
2022-02-27T14:10:15.7358175Z env:
2022-02-27T14:10:15.7358404Z   _R_CHECK_SYSTEM_CLOCK_: 0
2022-02-27T14:10:15.7358661Z   _R_CHECK_CRAN_INCOMING_REMOTE_: 0
2022-02-27T14:10:15.7358946Z   _R_CHECK_PKG_SIZES_THRESHOLD_: 100
2022-02-27T14:10:15.7359250Z ##[endgroup]
2022-02-27T14:11:54.2348646Z Downloading R and Rtools
2022-02-27T14:11:54.2404226Z Downloading https://cran.r-project.org/bin/windows/base/old/3.6.3/R-3.6.3-win.exe
2022-02-27T14:12:04.0057947Z Downloading https://github.com/microsoft/LightGBM/releases/download/v2.0.12/rtools35-x86_64.exe
2022-02-27T14:12:10.1839831Z Installing R
2022-02-27T14:12:49.7812622Z Done installing R
2022-02-27T14:12:49.7814678Z Installing Rtools
2022-02-27T14:14:33.8605607Z Done installing Rtools
2022-02-27T14:14:33.8606807Z Installing dependencies
2022-02-27T14:14:36.5274913Z package 'utf8' successfully unpacked and MD5 sums checked
2022-02-27T14:14:36.5794292Z package 'pillar' successfully unpacked and MD5 sums checked
2022-02-27T14:14:36.5928163Z package 'pkgconfig' successfully unpacked and MD5 sums checked
2022-02-27T14:14:36.6461005Z package 'vctrs' successfully unpacked and MD5 sums checked
2022-02-27T14:14:36.6600360Z package 'mime' successfully unpacked and MD5 sums checked
2022-02-27T14:14:36.6781763Z package 'glue' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.1627213Z package 'stringi' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.1772539Z package 'base64enc' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.1963509Z package 'rprojroot' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.2213568Z package 'rstudioapi' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.2535621Z package 'diffobj' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.2724167Z package 'fansi' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.2849508Z package 'rematch2' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.3282264Z package 'tibble' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.3411844Z package 'evaluate' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.3557043Z package 'highr' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.3828632Z package 'markdown' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.4039131Z package 'stringr' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.4256410Z package 'yaml' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.4455970Z package 'xfun' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.4692687Z package 'ps' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.4856722Z package 'htmltools' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.4978022Z package 'tinytex' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.5110265Z package 'brio' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.5251763Z package 'callr' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.5459448Z package 'cli' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.5579822Z package 'crayon' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.5788196Z package 'desc' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.6047237Z package 'digest' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.6176390Z package 'ellipsis' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.6402262Z package 'lifecycle' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.6642951Z package 'magrittr' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.6761674Z package 'pkgload' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.6865674Z package 'praise' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.7172257Z package 'rlang' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.7280116Z package 'waldo' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.7456043Z package 'withr' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.8489029Z package 'data.table' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.8770954Z package 'jsonlite' successfully unpacked and MD5 sums checked
2022-02-27T14:14:37.9860860Z package 'knitr' successfully unpacked and MD5 sums checked
2022-02-27T14:14:38.1098179Z package 'Matrix' successfully unpacked and MD5 sums checked
2022-02-27T14:14:38.1372197Z package 'processx' successfully unpacked and MD5 sums checked
2022-02-27T14:14:38.1487902Z package 'R6' successfully unpacked and MD5 sums checked
2022-02-27T14:14:38.3627724Z package 'rmarkdown' successfully unpacked and MD5 sums checked
2022-02-27T14:14:38.4409961Z package 'testthat' successfully unpacked and MD5 sums checked
2022-02-27T14:14:38.4437491Z 
2022-02-27T14:14:38.4438167Z The downloaded binary packages are in
2022-02-27T14:14:38.4438887Z 	C:\Users\runneradmin\AppData\Local\Temp\RtmpSwJ69r\downloaded_packages
2022-02-27T14:14:38.4803780Z Downloading https://github.com/microsoft/LightGBM/releases/download/v2.0.12/miktexsetup-4.0-x64.zip
2022-02-27T14:14:44.6379328Z Setting up MiKTeX
2022-02-27T14:14:58.0816413Z Installing MiKTeX
2022-02-27T14:16:03.8528264Z Done installing MiKTeX
2022-02-27T14:16:04.2126195Z initexmf: security risk: running with elevated privileges
2022-02-27T14:16:04.2141765Z 
2022-02-27T14:16:04.4514430Z Building R package
2022-02-27T14:16:06.8302815Z Building lightgbm with R executable: R
2022-02-27T14:16:08.6565334Z Removing files not needed for CRAN
2022-02-27T14:16:08.9282424Z Removing unknown pragmas in headers
2022-02-27T14:16:17.6945174Z Changing lib_lightgbm to lightgbm
2022-02-27T14:16:18.1849288Z Cleaning sed backup files
2022-02-27T14:16:18.4745412Z * checking for file 'lightgbm_r/DESCRIPTION' ... OK
2022-02-27T14:16:18.4748540Z 
2022-02-27T14:16:19.4143953Z * preparing 'lightgbm':
2022-02-27T14:16:19.4144793Z 
2022-02-27T14:16:19.4289335Z * checking DESCRIPTION meta-information ... OK
2022-02-27T14:16:19.4290147Z 
2022-02-27T14:16:19.4533112Z * cleaning src
2022-02-27T14:16:19.4533652Z 
2022-02-27T14:16:19.5970297Z * installing the package to build vignettes
2022-02-27T14:16:19.5970773Z 
2022-02-27T14:19:53.9150776Z * creating vignettes ... OK
2022-02-27T14:19:53.9151853Z 
2022-02-27T14:19:53.9490445Z * cleaning src
2022-02-27T14:19:53.9490909Z 
2022-02-27T14:19:54.2784402Z * checking for LF line-endings in source and make files and shell scripts
2022-02-27T14:19:54.2785694Z 
2022-02-27T14:19:55.2123182Z * checking for empty or unneeded directories
2022-02-27T14:19:55.2124427Z 
2022-02-27T14:19:55.2541130Z * looking to see if a 'data/datalist' file should be added
2022-02-27T14:19:55.2541793Z 
2022-02-27T14:19:55.2743361Z * building 'lightgbm_3.3.2.99.tar.gz'
2022-02-27T14:19:55.2743796Z 
2022-02-27T14:19:55.4260967Z Warning: file 'lightgbm/cleanup' did not have execute permissions: corrected
2022-02-27T14:19:55.4265407Z 
2022-02-27T14:19:55.4266360Z Warning: file 'lightgbm/configure' did not have execute permissions: corrected
2022-02-27T14:19:55.4266826Z 
2022-02-27T14:19:59.0452604Z 
2022-02-27T14:19:59.0452866Z 
2022-02-27T14:19:59.2061162Z removing object files created by vignettes
2022-02-27T14:19:59.3380405Z untarring lightgbm_3.3.2.99.tar.gz
2022-02-27T14:20:00.4079355Z done untarring lightgbm_3.3.2.99.tar.gz
2022-02-27T14:20:00.4080411Z re-tarring lightgbm_3.3.2.99.tar.gz
2022-02-27T14:20:00.5002054Z Running R CMD check
2022-02-27T14:20:01.0272005Z * using log directory 'C:/tmp-r-cmd-check/lightgbm.Rcheck'
2022-02-27T14:20:01.0272509Z 
2022-02-27T14:20:01.0274700Z * using R version 3.6.3 (2020-02-29)
2022-02-27T14:20:01.0279827Z 
2022-02-27T14:20:01.0281345Z * using platform: x86_64-w64-mingw32 (64-bit)
2022-02-27T14:20:01.0286397Z 
2022-02-27T14:20:01.0287692Z * using session charset: ISO8859-1
2022-02-27T14:20:01.0288008Z 
2022-02-27T14:20:01.0320023Z * checking package directory ... ERROR
2022-02-27T14:20:01.0322453Z 
2022-02-27T14:20:01.0331915Z package directory 'C:/tmp-r-cmd-check/lightgbm.Rcheck/00_pkg_src/lightgbm' does not exist
2022-02-27T14:20:01.0335281Z 
2022-02-27T14:20:01.0335674Z * DONE
2022-02-27T14:20:01.0335950Z 
2022-02-27T14:20:01.0937328Z 
2022-02-27T14:20:01.0937585Z 
2022-02-27T14:20:01.0945878Z Status: 1 ERROR
2022-02-27T14:20:01.0946735Z 
2022-02-27T14:20:01.0951808Z See
2022-02-27T14:20:01.0952119Z 
2022-02-27T14:20:01.0952782Z   'C:/tmp-r-cmd-check/lightgbm.Rcheck/00check.log'
2022-02-27T14:20:01.0953106Z 
2022-02-27T14:20:01.0953292Z for details.
2022-02-27T14:20:01.0953580Z 
2022-02-27T14:20:01.0954139Z 
2022-02-27T14:20:01.0954288Z 
2022-02-27T14:20:01.1177089Z R CMD check build logs:
2022-02-27T14:20:01.9827544Z [91mGet-Content: [0mD:\a\LightGBM\LightGBM\.ci\test_r_package_windows.ps1:198
2022-02-27T14:20:01.9828594Z [96mLine |
2022-02-27T14:20:01.9829168Z [96m 198 | [0m   [96mGet-Content -Path "$INSTALL_LOG_FILE_NAME"[0m
2022-02-27T14:20:01.9829804Z [96m     | [91m   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2022-02-27T14:20:01.9831510Z [91m[96m     | [91mCannot find path 'C:\tmp-r-cmd-check\lightgbm.Rcheck\00install.out' because it does not exist.
2022-02-27T14:20:01.9832246Z [0m
2022-02-27T14:20:02.0602081Z ##[error]Process completed with exit code 1.
```
https://github.com/microsoft/LightGBM/runs/5350688376?check_suite_focus=true

cc @jameslamb

--- a/.github/workflows/r_package.yml
+++ b/.github/workflows/r_package.yml
@@ -104,6 +104,12 @@ jobs:
           - os: windows-latest
             task: r-package
             compiler: MINGW
+            toolchain: MINGW
+            r_version: 3.6
+            build_type: cran
+          - os: windows-latest
+            task: r-package
+            compiler: MINGW
             toolchain: MSYS
             r_version: 4.1
             build_type: cran

--- a/build-cran-package.sh
+++ b/build-cran-package.sh
@@ -199,25 +199,33 @@ if ${BUILD_VIGNETTES} ; then
 
     echo "untarring ${TARBALL_NAME}"
     cd _tmp
-        tar -xvf "${TARBALL_NAME}" > /dev/null 2>&1
-        rm -rf "${TARBALL_NAME}"
-    cd ..
-    echo "done untarring ${TARBALL_NAME}"
+        tar -xf "${TARBALL_NAME}" > /dev/null 2>&1
+        rm -f "${TARBALL_NAME}"
+        echo "done untarring ${TARBALL_NAME}"
 
-    echo "re-tarring ${TARBALL_NAME}"
-    tar \
-        -czv \
-        -C ./_tmp \
-        --exclude=*.a \
-        --exclude=*.dll \
-        --exclude=*.o \
-        --exclude=*.so \
-        --exclude=*.tar.gz \
-        --exclude=**/conftest.c \
-        --exclude=**/conftest.exe \
-        -f "${TARBALL_NAME}" \
-        lightgbm \
-    > /dev/null 2>&1
+        # Object files are left behind from compiling the library to generate vignettes.
+        # Approaches like using tar --exclude=*.so to exclude them are not portable
+        # (for example, don't work with some versions of tar on Windows).
+        #
+        # Removing them manually here removes the need to use tar --exclude.
+        #
+        # For background, see https://github.com/microsoft/LightGBM/pull/3946#pullrequestreview-799415812.
+        rm -f ./lightgbm/src/*.o
+        rm -f ./lightgbm/src/boosting/*.o
+        rm -f ./lightgbm/src/io/*.o
+        rm -f ./lightgbm/src/metric/*.o
+        rm -f ./lightgbm/src/network/*.o
+        rm -f ./lightgbm/src/objective/*.o
+        rm -f ./lightgbm/src/treelearner/*.o
+
+        echo "re-tarring ${TARBALL_NAME}"
+        tar \
+            -cz \
+            -f "${TARBALL_NAME}" \
+            lightgbm \
+        > /dev/null 2>&1
+        mv "${TARBALL_NAME}" ../
+    cd ..
     echo "Done creating ${TARBALL_NAME}"
 
     rm -rf ./_tmp

